base/reduce.jl

```diff
@@ -579,11 +569,12 @@ true
 ```
 """
 any(f::Any, itr) = any(Predicate(f), itr)
-any(f::Predicate, itr) = mapreduce_sc_impl(f, |, itr)
-any(f::typeof(identity), itr) =
-    eltype(itr) <: Bool ?
-        mapreduce_sc_impl(f, |, itr) :
-        reduce(or_bool_only, false, itr)
+function any(f::Predicate, itr)
```

Comment:
Thinking about it a bit more, it would be even better to have definitions of `any` and `all` as the explicit loops, and then changing

```julia
mapreduce_sc_impl(f, ::typeof(|), itr) = any(f, itr)
mapreduce_sc_impl(f, ::typeof(&), itr) = all(f, itr)
```